pie/storage/__init__.py

```diff
@@ -0,0 +1,147 @@
+    """
+
+    db_value = StorageData.get(module.qualified_name, guild_id, key)
+    if not db_value:
+        return default_value
+
+    t = locate(db_value.type)
+
+    if not t:
+        return default_value
+
+    value = t(db_value.value)
+
+    return value
+
+
+def exists(module: commands.Cog, guild_id: int, key: str) -> bool:
+    """Checks if data for module, key and guild_id combination
+    are present in DB.
+
+    Args:
+        module (:class:`nextcord.ext.commands.Cog`): module connected with the value
+        guild_id (:class:'int'): ID of guild connected with the value (0 for global)
+        key (:class:'str'): Value's key
+
+        Returns:
+            True if value exists in DB, False otherwise
+    """
+
+    db_value = StorageData.get(module.qualified_name, guild_id, key)
+
+    return db_value is not None
+
+
+def get_type(module: commands.Cog, guild_id: int, key: str) -> type:
+    """Get data type.
+
+    Args:
+        module (:class:`nextcord.ext.commands.Cog`): module connected with the value
+        guild_id (:class:'int'): ID of guild connected with the value (0 for global)
+        key (:class:'str'): Value's key
+
+    Returns:
+        Value data type, None if type is not find
+    """
+
+    db_value = StorageData.get(module.qualified_name, guild_id, key)
+
+    if not db_value:
+        return None
+
+    t = locate(db_value.type)
+
+    return t
+
+
+def set(module: commands.Cog, guild_id: int, key: str, value: object) -> bool:
+    """Stores value into DB. If data exists, it's overwriten.
+
+    This is designed for basic data types (int, float, bool, string).
+    Using this for any other data types can cause problems!
+
+    Args:
+        module (:class:`nextcord.ext.commands.Cog`): module connected with the value
+        guild_id (:class:'int'): ID of guild connected with the value (0 for global)
+        key (:class:'str'): Value's key
+        value (:class: `typing.Any`): Value to store in DB
+
+    Returns:
+        True if succesfuly saved, False otherwise
+    """
+
+    return StorageData.set(module.qualified_name, guild_id, key, value) is not None
+
+
+def set_if_missing(module: commands.Cog, guild_id: int, key: str, value: Any) -> bool:
+    """Stores value into DB. If value exists, it's ignored.
+
+    This is designed for basic data types (int, float, bool, string).
+    Using this for any other data types can cause problems!
+
+    Args:
+        module (:class:`nextcord.ext.commands.Cog`): module connected with the value
+        guild_id (:class:'int'): ID of guild connected with the value (0 for global)
+        key (:class:'str'): Value's key
+        value (:class: `typing.Any`): Value to store in DB
+
+    Returns:
+        True if succesfuly saved, False otherwise
+    """
+
+    return (
+        StorageData.set(
+            module.qualified_name, guild_id, key, value, allow_overwrite=False
+        )
+        is not None
+    )
```

Comment:
ditto. Return the object if it was added and `None` otherwise.

Reply:
Same as previous - you should not get hands on StorageData object. The only right way to use it is through this endpoint.

Reply:
Same as above -- everything goes haywire if you both support `None` as valid value and compare it to `None` to return results.